src/main/java/teammates/ui/controller/PageData.java

```diff
@@ -1050,6 +1052,46 @@ public String getPictureUrl(String pictureKey) {
     public String removeBracketsForArrayString(String arrayString) {
         return arrayString.substring(1, arrayString.length() - 1).trim();
     }
+    
+    @SuppressWarnings("unused")
+    private void ___________methods_to_generate_comments() {
+    //========================================================================    
+    }
+    
+    public String getRecipientNames(Set<String> recipients, String courseId, String studentEmail, CourseRoster roster) {
+        StringBuilder namesStringBuilder = new StringBuilder();
+        int i = 0;
+        
+        for (String recipient : recipients) {
+            if (i == recipients.size() - 1 && recipients.size() > 1) {
+                namesStringBuilder.append(" and ");
+            } else if ( i > 0 && i < recipients.size() - 1 && recipients.size() > 2) {
+                namesStringBuilder.append(", ");
+            }
+            StudentAttributes student = roster.getStudentForEmail(recipient);
+            if (studentEmail != null && recipient.equals(studentEmail)) {
+                namesStringBuilder.append("you");
+            } else if (courseId.equals(recipient)) { 
+                namesStringBuilder.append("All Students In This Course");
```

Comment:
I think it's better to be "all students in this course" (all lowercase), especially since the way it's used is either "To ..." or "Comment about ...".